lib/cloud_controller/backends/runners.rb

```diff
@@ -73,31 +73,93 @@ def dea_apps(batch_size, last_id)
-          'state'         => app.state,
-          'memory'        => app.memory,
-          'version'       => app.version,
-          'updated_at'    => app.updated_at || app.created_at,
-          'package_state' => app.package_state,
-        }
+    def dea_apps_hm9k
+      # query 1
+      # get all process information where the process is STARTED and running on the DEA
+      process_query = App.db["Select p.id, p.app_guid, p.instances, p.version, apps.droplet_guid from processes p
+                      inner join apps ON (apps.guid = p.app_guid AND p.state ='STARTED' AND p.diego IS FALSE)"]
+      processes = process_query.all
+
+      # query 2
+      # get all necessary droplet information. This includes:
+      #    where the droplet's associated process is running on the DEA and the process is STARTED
+      #    Finding only the latest droplet associated with the process
+      droplets_query = App.db["select d.id, d.guid, d.app_guid, d.created_at, d.package_guid, d.state from droplets d
+                              join processes p ON (d.app_guid = p.app_guid AND p.state ='STARTED' AND p.diego IS FALSE)
+                              inner join (select app_guid, max(created_at) as _max from droplets group by app_guid) as x
+                              ON d.app_guid = x.app_guid and d.created_at=x._max"]
+      latest_droplets = latest(droplets_query.all)
+
+      # query 3
+      # get all necessary package information. This includes:
+      #   where the package's associated process is running on the DEA and the process is STARTED
+      #   finding only the latest package associated with the process
+      packages_query = App.db["select pkg.id, pkg.guid, pkg.app_guid, pkg.created_at, pkg.state from packages pkg
+                              join processes proc ON
+                                (pkg.app_guid = proc.app_guid AND proc.state ='STARTED' AND proc.diego IS FALSE)
+                              inner join (select app_guid, max(created_at) as _max from packages group by app_guid) as x
+                              ON pkg.app_guid = x.app_guid and pkg.created_at = x._max"]
+      latest_packages = latest(packages_query.all)
+
+      process_list = []
+      largest_id = 0
+
+      processes.each do |process|
+        app_guid = process[:app_guid]
+
+        pkg_state = package_state(app_guid, process[:droplet_guid], latest_droplets[app_guid], latest_packages[app_guid])
+        next if pkg_state == 'FAILED'
+
+        if process[:id] > largest_id
+          largest_id = process[:id]
+        end
+
+        process_list.push({
+          'id'            => app_guid,
+          'instances'     => process[:instances],
+          'state'         => 'STARTED',
+          'version'       => process[:version],
+          'package_state' => pkg_state,
+        })
+      end
+
+      [process_list, largest_id]
+    end
+
+    def latest(items)
+      current = {}
+
+      items.each do |item|
```

Comment:
You can also use https://ruby-doc.org/core-2.1.0/Enumerable.html#method-i-reduce instead of `each`